azurerm/data_source_batch_pool.go

```diff
@@ -0,0 +1,92 @@
+package azurerm
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func dataSourceArmBatchPool() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceArmBatchPoolRead,
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: azure.ValidateAzureRMBatchPoolName,
+			},
+			"resource_group_name": resourceGroupNameDiffSuppressSchema(),
+			"account_name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validateAzureRMBatchAccountName,
+			},
+			"display_name": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"vm_size": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"fixed_scale":             azure.SchemaBatchPoolFixedScaleForDataSource(),
+			"auto_scale":              azure.SchemaBatchPoolAutoScaleForDataSource(),
+			"storage_image_reference": azure.SchemaBatchPoolImageReferenceForDataSource(),
+			"node_agent_sku_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"start_task": azure.SchemaBatchPoolStartTaskForDataSource(),
+		},
+	}
+}
+
+func dataSourceArmBatchPoolRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).batchPoolClient
+
+	name := d.Get("name").(string)
+	accountName := d.Get("account_name").(string)
+	resourceGroup := d.Get("resource_group_name").(string)
+
+	ctx := meta.(*ArmClient).StopContext
+	resp, err := client.Get(ctx, resourceGroup, accountName, name)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			return fmt.Errorf("Error: Batch pool %q in account %q (Resource Group %q) was not found", name, accountName, resourceGroup)
+		}
+		return fmt.Errorf("Error making Read request on AzureRM Batch pool %q: %+v", name, err)
+	}
+
+	d.SetId(*resp.ID)
+
+	d.Set("name", name)
+	d.Set("account_name", accountName)
+	d.Set("resource_group_name", resourceGroup)
+	d.Set("vm_size", resp.VMSize)
```

Comment:
The Azure API returns the representation of a resource at the last time it was modified; meaning if it’s not been modified in a while (for example if it was made during a Preview API) the structure returned from the API may not necessarily match the result we expect. As such we ensure we access these fields in the Properties block, since that’s where we expect them